### PR TITLE
documentation #800: updated swagger doc for role endpoints:

### LINKED ISSUE #800: Update Swagger doc for Role (2)
We are updating the documentation for all endpoints and adding more _description and details_. 

One of the main controllers is Role. You should update the documentation according to the **pattern** below:
```ts
  @ApiBearerAuth()
  @ApiOkResponse()
  @ApiBadRequestResponse({
    description: `\n
    InvalidEntityIdException:
      User with such id is not found

    AlreadyRegisteredException:
      User is already registered

    InvalidBodyException:
      Group id can not be empty
      IsCaptain must be a boolean
      IsCaptain can not be empty
    
    AbsenceOfCaptainException:
      Captain was not found
    
    AbsenceOfCaptainTelegramException:
      Captain's telegramId was not found`,
  })
  @ApiUnauthorizedResponse({
    description: `\n
    UnauthorizedException:
      Unauthorized`,
  })
  @ApiForbiddenResponse({
    description: `\n
    NoPermissionException:
      You do not have permission to perform this action
    
    ForbiddenException:
      Forbidden`,
  })
  @ApiParam({
    name: 'userId',
    required: true,
    description: 'Id of a user to request to the group',
  })
  @ApiEndpoint({
    summary: 'Request captain or admin to join the group',
    permissions: PERMISSION.USERS_$USERID_GROUP_REQUEST,
  })
  @Patch('/:userId/requestNewGroup')
  requestNewGroup (
    @Param('userId', UserByIdPipe) userId: string,
    @Body() body: GroupRequestDTO,
  ) {
    return this.userService.requestNewGroup(userId, body);
  }
```
**Structure:**
- auth (if needed)
- ok response
- client errors
- params/query description
- api endpoint for summary, permissions, guards

You should also add **field description** in the bady or response.
Add **summary** to endpoints (fix this issue).
Set **default values** for the optional fields

image

**Example**: #777

_My DM is always open for you. Happy coding_ 🐸

[Our Swagger](https://apidev.fictadvisor.com/api)
[Doc](https://docs.nestjs.com/openapi/introduction)

### PR DESCRIPTION
**Pull request by Dmutre**
- POST /v2/roles/{roleId}/grants
- GET v2/roles/{roleId}/grants

closes #800

### CODE REVIEW

File: src/v2/api/controllers/RoleController.ts [hunk 1]
```diff
@@ -121,11 +125,14 @@ export class RoleController {
...
-      Permission cannot be empty
-      Set is not a boolean`,
+  @ApiParam({
+    name: 'roleId',
+    required: true,
+    description: 'Id of the role to which you want to give grants',
+  })
+  @ApiEndpoint({
+    summary: 'Give grants to the role by id',
+    permissions: PERMISSION.ROLES_$ROLEID_GRANTS_CREATE,
```
Comment (SharpDevOps10) [inline src/v2/api/controllers/RoleController.ts:135]: You should also add `description` field to `CreateGrantsDTO`

File: src/v2/api/controllers/RoleController.ts [hunk 2]
```diff
@@ -106,11 +105,16 @@ export class RoleController {
...
   @ApiBearerAuth()
   @ApiOkResponse({
     type: GrantResponse,
   })
+  @ApiBadRequestResponse({
+    description: `\n
+    InvalidBodyException:
+      Permission cannot be empty
+      Set must be a boolean`,
+  })
```
Comment (SharpDevOps10) [inline src/v2/api/controllers/RoleController.ts:117]: You should also add `description` field to `GrantResponse`

File: src/v2/api/controllers/RoleController.ts [hunk 3]
```diff
@@ -138,6 +145,14 @@ export class RoleController {
   @ApiOkResponse({
     type: MappedGrantsResponse,
   })
```
Comment (SharpDevOps10) [inline src/v2/api/controllers/RoleController.ts:147]: You should also add `description` field to `MappedGrantsResponse`

File: src/v2/api/controllers/RoleController.ts [hunk 4]
```diff
@@ -106,11 +105,16 @@ export class RoleController {
...
-  @Access(PERMISSION.ROLES_$ROLEID_GRANTS_CREATE)
   @ApiBearerAuth()
   @ApiOkResponse({
     type: GrantResponse,
   })
+  @ApiBadRequestResponse({
+    description: `\n
+    InvalidBodyException:
+      Permission cannot be empty
+      Set must be a boolean`,
```
Comment (SharpDevOps10) [inline src/v2/api/controllers/RoleController.ts:116]: Return `Set is not a boolean`. Check `CreateGrantDTO`

File: src/v2/api/controllers/RoleController.ts [hunk 5]
```diff
@@ -182,6 +197,12 @@ export class RoleController {
   @ApiOkResponse({
     type: BaseRoleResponse,
   })
+  @ApiBadRequestResponse({
+    description: `\n
+    InvalidBodyException:
+      Name must be an enum
+      Weight must be a number`,
+  })
```
Comment (SharpDevOps10) [inline src/v2/api/controllers/RoleController.ts:205]: Return previous exceptions. Check `UpdateRoleDTO`

File: src/v2/api/controllers/RoleController.ts [hunk 6]
```diff
@@ -70,6 +69,19 @@ export class RoleController {
...
+    InvalidEntityIdException: 
+      ParentId with such id was not found
+      
+    InvalidBodyException:
+      Name must be an enum
+      Name cannot be empty
+      Weight must be a number 
+      Weight cannot be empty 
+      Permission cannot be empty 
+      Set must be a boolean`,
```
Comment (SharpDevOps10) [inline src/v2/api/controllers/RoleController.ts:83]: Return previous exceptions. Check `CreateRoleDTO`

**Review by SharpDevOps10 — APPROVED** [review]

**Review by fujivara — APPROVED** [review]

**Review by fokaaas — APPROVED** [review]

**Pull request merged**